### ISSUE TITLE
confused about  img_corrected

### ISSUE THREAD

**Issue by zyqgmzyq** [yuanming-hu/fc4#23]
img_corrected = tf.pow(
      images[:, :, :, ::-1] / 65535 / illums_pooled[:, None, None, :] *
      tf.reduce_mean(illums_pooled, axis=(1), keep_dims=True)[:, None, None, :],
      1 / VIS_GAMMA)

Why need to multipy np.mean(illums_pooled)?

**Comment by yuanming-hu** [comment]
Hi, thanks for the question.

This is just to normalize the estimation. Note that color constancy has only two DoFs instead of three (see, FAQ (h)).

**Comment by zyqgmzyq** [comment]
If I want to use the ground-truth to correct the picture, 

img_corrected = np.power(mages[:, :, :, ::-1] / 65535 / illums_ground_truth , 1/2.2), is it right？

**Comment by zyqgmzyq** [comment]
It seems so complex, I'm not clearly understand about that. Now I want to visualize the raw image and corrected image after divided the image by illums_ground_truth , Is there some standard operation to corrected the image ? 

I just divide the image by illums_ground_truth, but the image is dark. Your mean is that this image is not exposure? So the raw image is not exposure?  So visualize the corrected image not only should remove the color cast but need to exposure?

**Comment by zyqgmzyq** [comment]
It seems so complex, I'm not clearly understand about that. Now I want to visualize the raw image and corrected image after divided the image by illums_ground_truth , Is there some standard operation to corrected the image ?

I just divide the image by illums_ground_truth, but the image is dark. Your mean is that this image is not exposure? So the raw image is not exposure? So visualize the corrected image not only should remove the color cast but need to exposure?

**Comment by yuanming-hu** [comment]
You are right that color constancy only takes care of the color cast, not the exposure. However, when visualizing them people also want the images to be correctly exposed. Therefore, you do not want `illums_pooled` to be a very "long" vector, since that may make the image very dark. That's why you normalize this vector, or multiple the image the length of the vector after dividing it.

**Comment by yuanming-hu** [comment]
A common standard is to scale the RGB values so that their average is 0.18
(and then do the 1/2.2 gamma).

Best,
Yuanming
--------------------------------------------------------------
MIT CSAIL

**Comment by zyqgmzyq** [comment]
Sorry，I don't know what RGB value is.   Could you please describe it in detail?
And would you recommend some related materials or papers ？
Thanks!

**Comment by zyqgmzyq** [comment]
Sorry，I don't know what RGB value is. Could you please describe it in detail?
And would you recommend some related materials or papers ？
Thanks!
